ipv6 structures in <netinet/in.h> are used by the <sys/socket.h> so it should be included first
ipv6 structures needs explicit CFLAGS " -D_GNU_SOURCE" to compile on linux
diff -ru dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.orig/include/connection.h dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.new/include/connection.h
--- dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.orig/include/connection.h	2018-03-21 04:21:30.193354693 +0100
+++ dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.new/include/connection.h	2018-03-21 04:46:46.450694148 +0100
@@ -28,11 +28,11 @@
 #ifndef HAVE_CONNECTION_H
 #define HAVE_CONNECTION_H
 
+#include <netinet/in.h>
 #include <sys/socket.h>
 #include <stdbool.h>
 #include <stdint.h>
 
-#include <netinet/in.h>
 //#include <linux/if.h>
 #include <ev.h>
 #include <glib.h>
